internal/utils/utils.go

```diff
@@ -27,8 +28,14 @@ func BuildImagePath(userImage, defaultImage, userVersion, defaultVersion string)
 		userImage = defaultImage
 	}
 
+	var checkNumber = regexp.MustCompile(`^[1-9].*`).MatchString
```

Comment:
I think we should be a little more strict here, maybe something more like in this link:

https://github.com/cilium/cilium/blob/006140eca1950690de5cf310219ce6f10c5aa656/contrib/release/lib/gitlib.sh#L42

Note though the regex syntax might not be 100% the same.

What I have in mind is we check something that may or may not have a beginning `v`, followed by a semantic version (also supporting values like `v1.10.0-rc1`). If this doesn't pass, then the version is nonsense.

If the regex is OK, we can do something like `strings.StartsWith(userVersion, "v")` to check whether there is a v in front, if it's not there, just prepend it. If it's there, nothing to do - can just continue using the value.